website/source/docs/guides/segments.html.markdown.erb

```diff
@@ -0,0 +1,229 @@
+  via network rules or firewalls. Prior to Consul 0.9.3 this was only possible through federation,
+  which for some users is too heavyweight or expensive as it requires running multiple servers per
+  segment.
+---
+
+# Partial LAN Connectivity
+## Configuring Network Segments
+
+[//]: # ( ~> The network area functionality described here is available only in )
+[//]: # (   [Consul Enterprise](https://www.hashicorp.com/products/consul/) version 0.8.0 and later. )
+
+<%= enterprise_alert :consul %>
+
+Many advanced Consul users have the need to run clusters with segmented networks, meaning that
+not all agents can be in a full mesh. This is usually the result of business policies enforced
+via network rules or firewalls. Prior to Consul 0.9.3 this was only possible through federation,
+which for some users is too heavyweight or expensive as it requires running multiple servers per
+segment.
+
+By default, all Consul agents in one datacenter are part of a shared gossip pool over the LAN;
+this means that the partial connectivity caused by segmented networks would cause health flapping
+as nodes failed to communicate. In this guide we will cover the Network Segments feature, added
+in [Consul Enterprise](https://www.hashicorp.com/products/consul/) version 0.9.3, which allows users
+to configure Consul to support this kind of segmented network topology.
+
+This guide will cover the basic configuration for setting up multiple segments, as well as
+how to configure a prepared query to limit service discovery to the services in the local agent's
+network segment.
+
+## Network Segments
+
+All Consul agents are part of the default network segment, `""`, unless a segment is specified in
+their configuration. In a standard cluster setup all agents will normally be part of this default
+segment and as a result, part of one shared LAN gossip pool. Network segments can be used to break
+up the LAN gossip pool into multiple isolated smaller pools by specifying the configuration for segments
+on the servers. Each desired segment must be given a name and port, as well as optionally a custom
+bind and advertise address for that segment's gossip listener to bind to on the server.
+
+A few things to note:
+1. Servers will be a part of all segments they have been configured with. They are the common point
+linking the different segments together.
+2. Client agents can only be part of one segment at a given time, specified by the [`-segment`]
+(/docs/agent/options.html#_segment) option.
+3. Clients can only join agents in the same segment as them. If they attempt to join a client in
+another segment, or the listening port of another segment on a server, they will get a segment mismatch error.
+
+Once the servers have been configured with the correct segment info, the clients only need to specify
+their own segment in the [Agent Config](/docs/agent/options.html) and join by connecting to another
+agent within the same segment.
```

Comment:
We should add something like "If joining to a Consul server, client will need to provide the server's port for their segment along with the address of the server when performing the join (eg. `consul agent -retry-join "consul.domain.internal:1234"`).